airflow/www/templates/airflow/calendar.html

```diff
@@ -0,0 +1,63 @@
+{#
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+#}
+
+{% extends "airflow/dag.html" %}
+{% block page_title %}{{ dag.dag_id }} - Tree - {{ appbuilder.app_name }}{% endblock %}
+
+{% block head_css %}
+  {{ super() }}
+  <meta name="dag_id" content="{{ dag.dag_id }}">
+  <meta name="tree_url" content="{{ url_for('Airflow.tree') }}">
+  <style type="text/css">
+    {% for state, state_color in state_color_mapping.items() %}
+      rect.{{state}} {
+        fill: {{state_color}} !important;
+      }
+    {% endfor %}
+  </style>
+{% endblock %}
+
+{% block content %}
+  {{ super() }}
+  <hr>
+  <div class="legend-row">
+    <div>
+      {% for state, state_color in state_color_mapping.items() %}<span class="legend-item legend-item--no-border">
+        <span class="legend-item__swatch legend-item__swatch--no-border" style="background: {{ state_color }};"></span>
+        {{state}}</span>{% endfor %}<span class="legend-item legend-item--no-border">
+        <span class="legend-item__swatch"></span>no_status</span>
+    </div>
+  </div>
+  <hr>
+  <div id="svg_container">
+    <img id='loading' width="50" src="{{ url_for('static', filename='loading.gif') }}">
+    <svg id="calendar-svg">
+    </svg>
+  </div>
+{% endblock %}
+
+{% block tail_js %}
+  {{ super() }}
+  <script src="{{ url_for_asset('d3.min.js') }}"></script>
+  <script src="{{ url_for_asset('d3-tip.js') }}"></script>
+  <script src="{{ url_for_asset('calendar.js') }}"></script>
+  <script>
+    const calendarData = {{ data|tojson }};
+  </script>
```

Comment:
You'll want to use a similar pattern to what you'll see towards the top of this file to pass this data:
```html
<meta name="calendar_data" content="{{ data|tojson }}">
```